src/main/java/reposense/RepoSense.java

```diff
@@ -65,11 +65,17 @@ public static void main(String[] args) {
             RepoConfiguration.setDatesToRepoConfigs(configs, cliArguments.getSinceDate(), cliArguments.getUntilDate());
             RepoConfiguration.setStandaloneConfigIgnoredToRepoConfigs(configs,
                     cliArguments.isStandaloneConfigIgnored());
+
             List<Path> reportFoldersAndFiles = ReportGenerator.generateReposReport(configs,
+                    cliArguments.getOutputFilePath().toAbsolutePath().toString());
+
+            reportFoldersAndFiles.addAll(ReportGenerator.generateSummary(configs,
```

Comment:
doesn't quite make sense to split it into 2; this should be the underlying structure.
Think of it from another angle, shouldn't the `generateReposReport` comes with a summary?

Reply:
Thanks for the feedback. I have reverted to the original `generateReposReport` method.
My original reason for splitting this method is to allow the test code to provide its own report generation time.

However, this can be done by accepting a supplier for the report generation time, as done in the current solution.